sdk/test/UnitTests/Custom/Marshalling/StreamingUtf8JsonReaderTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using AWSSDK_DotNet.CommonTest.Utils;
+using AWSSDK_DotNet.UnitTests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Amazon.Runtime.Internal.Util;
+using Amazon.EC2.Model;
+namespace AWSSDK.UnitTests
+{
+    /// <summary>
+    /// Protocol tests already exists to test the marshalling and unmarshalling of request and responses in json, but they don't test very
+    /// large payloads, which would trigger the logic for getting more bytes from the stream. 
+    /// This class just tests the wrapper class StreamingUtf8JsonReader. 
+    /// </summary>
+    [TestClass]
+    public class StreamingUtf8JsonReaderTests
+    {
+        [TestMethod]
+        public void HandlesUtf8BOM()
+        {
+            // we can't use reflection to access the private fields of StreamingUtf8JsonReader since it is a ref struct so we have to test it this way.
```

Comment:
You could make the methods `internal` in StreamingUtf7JsonReader` and then since the unit test project has access via the `InternalsVisibleToAttribute` test the handle method directly. Following this pattern you could also have tests directly on `FillBuffer` and `GetMoreBytesFromStream`,

Reply:
I see the value in testing FillBuffer directly, but I don't really see the value in testing `GetMoreBytesFromStream` directly since this method is only called in the context of `Read()`. The test where I make one Json token = defaultvalueBufferSize + 500 covers the array resizing logic, and the smaller buffer size covers the `reader.BytesConsumed < buffer.Length` logic. However, I could add one additional test where the stream is exactly the size of the buffer.

So I' like to keep as many methods private as I can, but if you feel strongly about this I can change it to internal. I just don't see what additional testing I would do other than exact buffer size.